framework/src/packets/ip/v6/srh.rs

```diff
@@ -193,7 +207,7 @@ impl<E: Ipv6Packet> SegmentRouting<E> {
             self.header_mut().segments_left = segments_left;
             self.envelope_mut().set_dst(IpAddr::V6(segment))
         } else {
-            Err(SegmentsLeftOutOfBoundError(segments_left, self.segments().len()).into())
+            Err(SegmentsLeftOutOfBoundsError(segments_left, self.segments().len()).into())
```

Comment:
can you update the code to use the extension trait's `current_segment`?  and that method can indeed err with malformed data.

Reply:
not sure what I was thinking here. ignore this comment.